### ISSUE TITLE
[VersionControl] Crash rebasing (Git)

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7110]
Trying to rebase fails sometimes (not always) with an IDE crash.

```
  at <unknown> <0xffffffff>
  at LibGit2Sharp.Core.NativeMethods:git_stash_save <0x00144>
  at LibGit2Sharp.Core.Proxy:git_stash_save <0x000ca>
  at LibGit2Sharp.StashCollection:Add <0x00122>
  at MonoDevelop.VersionControl.Git.GitRepository:TryCreateStash <0x000f7>
  at MonoDevelop.VersionControl.Git.GitRepository:CommonPreMergeRebase <0x00442>
  at MonoDevelop.VersionControl.Git.GitRepository:Rebase <0x00102>
  at MonoDevelop.VersionControl.Git.GitService:ShowMergeDialog <0x00272>
  at MonoDevelop.VersionControl.Git.RebaseBranchHandler:Run <0x00032>
  at MonoDevelop.Components.Commands.CommandHandler:Run <0x00013>
  at MonoDevelop.Components.Commands.CommandHandler:InternalRun <0x00018>
  at MonoDevelop.Components.Commands.CommandManager:DefaultDispatchCommand <0x00292>
  at MonoDevelop.Components.Commands.CommandManager:DispatchCommand <0x01282>
  at MonoDevelop.Components.Commands.CommandManager:DispatchCommand <0x0005a>
  at <>c__DisplayClass14_0:<Run>b__0 <0x000e2>
  at TimeoutProxy:HandlerInternal <0x0007c>
  at TimeoutProxy:HandlerInternal <0x00062>
  at <unknown> <0xffffffff>
  at Gtk.Application:gtk_main <0x00056>
  at Gtk.Application:Run <0x00012>
  at MonoDevelop.Ide.IdeApp:Run <0x00022>
  at MonoDevelop.Ide.IdeStartup:Run <0x03402>
  at MonoDevelop.Ide.IdeStartup:Main <0x003d2>
  at MonoDevelop.Startup.MonoDevelopMain:Main <0x0001a>
  at <Module>:runtime_invoke_int_object <0x0010c>
```

Regression #7078

> VS bug [#789307](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789307)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1594 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)
